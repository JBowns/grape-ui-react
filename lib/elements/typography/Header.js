'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Header = undefined;

var _templateObject = _taggedTemplateLiteral(['\n    ', '\n  '], ['\n    ', '\n  ']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _textStyles = require('./textStyles');

var _fontSize = require('../../assets/json/fontSize.json');

var fontSizeObject = _interopRequireWildcard(_fontSize);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Header = function Header(props) {
  var h2 = props.h2,
      h3 = props.h3,
      h4 = props.h4,
      h5 = props.h5,
      h6 = props.h6,
      otherProps = _objectWithoutProperties(props, ['h2', 'h3', 'h4', 'h5', 'h6']);

  var headingElement = 'h1';
  var headingFontSize = fontSizeObject.h1;
  if (h2) {
    headingElement = 'h2';
    headingFontSize = fontSizeObject.h2;
  } else if (h3) {
    headingElement = 'h3';
    headingFontSize = fontSizeObject.h3;
  } else if (h4) {
    headingElement = 'h4';
    headingFontSize = fontSizeObject.h4;
  } else if (h5) {
    headingElement = 'h5';
    headingFontSize = fontSizeObject.h5;
  } else if (h6) {
    headingElement = 'h6';
    headingFontSize = fontSizeObject.h6;
  }
  var overrides = Object.assign({}, props, {
    fontSizeBase: headingFontSize
  });
  var actualBase = (0, _textStyles.textStylesBase)(overrides);
  var ProtoHeader = _styledComponents2.default[headingElement](_templateObject, actualBase);
  return _react2.default.createElement(ProtoHeader, otherProps);
};

exports.Header = Header;
Header.propTypes = {
  h2: _propTypes2.default.bool,
  h3: _propTypes2.default.bool,
  h4: _propTypes2.default.bool,
  h5: _propTypes2.default.bool,
  h6: _propTypes2.default.bool
};

Header.defaultProps = {
  h2: false,
  h3: false,
  h4: false,
  h5: false,
  h6: false
};