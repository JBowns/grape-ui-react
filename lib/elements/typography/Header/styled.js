"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty2 = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty2(exports, "__esModule", {
  value: true
});

exports.Header = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/define-property"));

var _defineProperties = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/define-properties"));

var _getOwnPropertyDescriptors = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/get-own-property-descriptors"));

var _forEach = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/for-each"));

var _getOwnPropertyDescriptor = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/get-own-property-descriptor"));

var _filter = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/filter"));

var _getOwnPropertySymbols = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/get-own-property-symbols"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/taggedTemplateLiteral"));

var _keys = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/keys"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/objectWithoutProperties"));

var _defineProperty3 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _styledHelpers = require("../../../utils/styledHelpers");

var _globalStyles = require("../../../global-styles");

var _componentHelpers = require("../../../utils/componentHelpers");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function ownKeys(object, enumerableOnly) { var keys = (0, _keys["default"])(object); if (_getOwnPropertySymbols["default"]) { var symbols = (0, _getOwnPropertySymbols["default"])(object); if (enumerableOnly) symbols = (0, _filter["default"])(symbols).call(symbols, function (sym) { return (0, _getOwnPropertyDescriptor["default"])(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { var _context; (0, _forEach["default"])(_context = ownKeys(Object(source), true)).call(_context, function (key) { (0, _defineProperty3["default"])(target, key, source[key]); }); } else if (_getOwnPropertyDescriptors["default"]) { (0, _defineProperties["default"])(target, (0, _getOwnPropertyDescriptors["default"])(source)); } else { var _context2; (0, _forEach["default"])(_context2 = ownKeys(Object(source))).call(_context2, function (key) { (0, _defineProperty2["default"])(target, key, (0, _getOwnPropertyDescriptor["default"])(source, key)); }); } } return target; }

var _getGlobalStyles = (0, _globalStyles.getGlobalStyles)(),
    fontSizeSchema = _getGlobalStyles.fontSize,
    gridSchema = _getGlobalStyles.grid;

var getHeaderFontSizeFromTag = function getHeaderFontSizeFromTag(factoryProps) {
  var factoryPropsTag = factoryProps.tag;
  var tag = factoryPropsTag || 'h1';

  var getHeaderFontSizeMemoized = function getHeaderFontSizeMemoized(props) {
    var displayHeader = props.displayHeader;

    var overrides = _objectSpread({}, props, {
      fontSize: displayHeader ? fontSizeSchema.displayHeader[tag] : fontSizeSchema[tag]
    });

    return (0, _styledHelpers.fontSizeCore)(overrides);
  };

  return getHeaderFontSizeMemoized;
};

var getHeaderFontWeight = function getHeaderFontWeight(props) {
  var displayHeader = props.displayHeader,
      fontWeightFromProps = props.fontWeight;

  var overrides = _objectSpread({}, props, {
    fontWeight: displayHeader ? '300' : fontWeightFromProps || _styledHelpers.defaultStylesBase.fontWeight
  });

  return (0, _styledSystem.fontWeight)(overrides);
};

var propsToTrim = _objectSpread({}, _styledHelpers.spaceProps.propTypes, {}, _styledHelpers.typography.propTypes, {
  children: _propTypes["default"].any.isRequired,
  displayHeader: _propTypes["default"].bool
});

var headerFactory = function headerFactory(factoryProps) {
  var tag = factoryProps.tag;

  var HeaderComponent = function HeaderComponent(_ref) {
    var children = _ref.children,
        props = (0, _objectWithoutProperties2["default"])(_ref, ["children"]);
    return _react["default"].createElement(tag, (0, _componentHelpers.removeSomeProps)(props, (0, _keys["default"])(propsToTrim)), children);
  }; // This is stated here strictly as a interface reference. Unfortunately as this component is dynamically created there is no propType required or defaultProps enforced.


  HeaderComponent.propTypes = {
    children: _propTypes["default"].any.isRequired
  };
  var getHeaderFontSize = getHeaderFontSizeFromTag(factoryProps);
  return (0, _styledComponents["default"])(HeaderComponent)(_templateObject(), _styledHelpers.colorCore, _styledHelpers.ellipsisCore, _styledHelpers.fontFamilyCore, _styledHelpers.fontStyleCore, getHeaderFontSize, getHeaderFontWeight, _styledHelpers.letterSpacingCore, _styledHelpers.lineHeightCore, _styledHelpers.textAlignCore, _styledHelpers.textDecorationCore, _styledSystem.space);
};

var Header = headerFactory({
  tag: 'h1'
});
exports.Header = Header;
Header.h1 = Header;

for (var i = 2; i <= 6; i++) {
  var subHeaderTag = "h".concat(i);
  Header[subHeaderTag] = headerFactory({
    tag: subHeaderTag
  });
  Header[subHeaderTag].propTypes = _objectSpread({
    displayHeader: _propTypes["default"].bool
  }, _styledHelpers.typography.propTypes);
  Header[subHeaderTag].defaultProps = _objectSpread({}, _styledHelpers.defaultStylesBase, {
    displayHeader: false,
    fontWeight: _styledHelpers.defaultStylesBase.fontWeight,
    mb: gridSchema.gutter,
    mt: 0
  });
}

Header.propTypes = _objectSpread({}, _styledHelpers.typography.propTypes, {
  displayHeader: _propTypes["default"].bool
});
Header.defaultProps = _objectSpread({}, _styledHelpers.defaultStylesBase, {
  displayHeader: false,
  fontWeight: _styledHelpers.defaultStylesBase.fontWeight,
  mb: gridSchema.gutter,
  mt: 0
});
/** @component */