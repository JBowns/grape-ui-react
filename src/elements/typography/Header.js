import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { textStylesBase } from './textStyles';
import * as fontSizeObject from '../../assets/json/fontSize.json';

export const Header = props => {
  const { h2, h3, h4, h5, h6, ...otherProps } = props;
  let headingElement = 'h1';
  let headingFontSize = fontSizeObject.h1;
  if (h2) {
    headingElement = 'h2';
    headingFontSize = fontSizeObject.h2;
  } else if (h3) {
    headingElement = 'h3';
    headingFontSize = fontSizeObject.h3;
  } else if (h4) {
    headingElement = 'h4';
    headingFontSize = fontSizeObject.h4;
  } else if (h5) {
    headingElement = 'h5';
    headingFontSize = fontSizeObject.h5;
  } else if (h6) {
    headingElement = 'h6';
    headingFontSize = fontSizeObject.h6;
  }
  const overrides = {
    ...props,
    fontSizeBase: headingFontSize,
  };
  const actualBase = textStylesBase(overrides);
  const ProtoHeader = styled[headingElement]`
    ${actualBase}
  `;
  return <ProtoHeader {...otherProps} />;
};

Header.propTypes = {
  h2: PropTypes.bool,
  h3: PropTypes.bool,
  h4: PropTypes.bool,
  h5: PropTypes.bool,
  h6: PropTypes.bool,
};

Header.defaultProps = {
  h2: false,
  h3: false,
  h4: false,
  h5: false,
  h6: false,
};
