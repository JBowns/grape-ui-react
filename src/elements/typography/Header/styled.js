import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { fontWeight } from 'styled-system';
import {
  colorCore,
  defaultStylesBase,
  fontFamilyCore,
  fontSizeCore,
  fontStyleCore,
  letterSpacingCore,
  lineHeightCore,
  textAlignCore,
  textDecorationCore,
  typography,
} from 'src/utils/styledHelpers';
import { getGlobalStyles } from 'src/global-styles';
import { removeSomeProps } from 'src/utils/componentHelpers';

const { fontSize: fontSizeSchema, grid: gridSchema } = getGlobalStyles();

const getHeaderFontSizeFromTag = factoryProps => {
  const { tag: factoryPropsTag } = factoryProps;
  const tag = factoryPropsTag || 'h1';
  const getHeaderFontSizeMemoized = props => {
    const { display } = props;
    const overrides = {
      ...props,
      fontSize: display ? fontSizeSchema.display[tag] : fontSizeSchema[tag],
    };
    return fontSizeCore(overrides);
  };
  return getHeaderFontSizeMemoized;
};

const getHeaderFontWeight = props => {
  const {
    display, fontWeight: fontWeightFromProps,
  } = props;
  const overrides = {
    ...props,
    fontWeight: display ? '300' : fontWeightFromProps || defaultStylesBase.fontWeight,
  };
  return fontWeight(overrides);
};
const propsToTrim = {
  children: PropTypes.any.isRequired,
  ...typography.propTypes,
  display: PropTypes.bool,
};

const headerFactory = factoryProps => {
  const { tag } = factoryProps;
  const HeaderComponent = ({
    children, ...props
  }) => React.createElement(tag, removeSomeProps(props, Object.keys(propsToTrim)), children);
  // This is stated here strictly as a interface reference. Unfortunately as this component is dynamically created there is no propType required or defaultProps enforced.
  HeaderComponent.propTypes = {
    children: PropTypes.any.isRequired,
  };
  const getHeaderFontSize = getHeaderFontSizeFromTag(factoryProps);
  return styled(HeaderComponent)`
    ${colorCore}
    ${fontFamilyCore}
    ${fontStyleCore}
    ${getHeaderFontSize}
    ${getHeaderFontWeight}
    ${letterSpacingCore}
    ${lineHeightCore}
    ${textAlignCore}
    ${textDecorationCore}
    margin: 0 0 ${gridSchema.gutter};
  `;
};

const Header = headerFactory({ tag: 'h1' });

Header.h1 = Header;
Header.h2 = headerFactory({ tag: 'h2' });
Header.h3 = headerFactory({ tag: 'h3' });
Header.h4 = headerFactory({ tag: 'h4' });
Header.h5 = headerFactory({ tag: 'h5' });
Header.h6 = headerFactory({ tag: 'h6' });

Header.propTypes = {
  display: PropTypes.bool,
  ...typography.propTypes,
};

Header.defaultProps = {
  display: false,
  fontWeight: defaultStylesBase.fontWeight,
};
export { Header };
