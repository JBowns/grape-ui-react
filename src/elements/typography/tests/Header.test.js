import { Header } from '../Header';

describe('Header Component', () => {
  it('should return a Header object', () => {
    expect(Header).toBeTruthy();
  });
});
