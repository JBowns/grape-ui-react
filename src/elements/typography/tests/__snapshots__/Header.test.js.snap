// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Elements Component should return a Header h2 object 1`] = `
.c0 {
  font-size: calc(2em * 1);
  line-height: 1.5;
}

<h2
  className="c0"
>
  Lorem Ipsum
</h2>
`;

exports[`Header Elements Component should return a Header h3 object 1`] = `
.c0 {
  font-size: calc(1.75em * 1);
  line-height: 1.5;
}

<h3
  className="c0"
>
  Lorem Ipsum
</h3>
`;

exports[`Header Elements Component should return a Header h4 object 1`] = `
.c0 {
  font-size: calc(1.5em * 1);
  line-height: 1.5;
}

<h4
  className="c0"
>
  Lorem Ipsum
</h4>
`;

exports[`Header Elements Component should return a Header h5 object 1`] = `
.c0 {
  font-size: calc(1.25em * 1);
  line-height: 1.5;
}

<h5
  className="c0"
>
  Lorem Ipsum
</h5>
`;

exports[`Header Elements Component should return a Header h6 object 1`] = `
.c0 {
  font-size: calc(1em * 1);
  line-height: 1.5;
}

<h6
  className="c0"
>
  Lorem Ipsum
</h6>
`;

exports[`Header.h1 Component should return a Header object with font-size: lg 1`] = `
.c0 {
  font-size: calc(2.5em * 1.2);
  line-height: 1.5;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component should return a Header object with font-size: sm 1`] = `
.c0 {
  font-size: calc(2.5em * .8);
  line-height: 1.5;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component should return a Header object with font-style: base 1`] = `
.c0 {
  font-size: calc(2.5em * 1);
  line-height: 1.5;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component should return a Header object with font-style: italic 1`] = `
.c0 {
  font-size: calc(2.5em * 1);
  line-height: 1.5;
  font-style: italic;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;
